test/ctia/stores/es/crud_test.clj

```diff
@@ -581,3 +607,105 @@
+                                              (map :id tested-sample)
+                                              {:limit total})]
+           (is (= (set tested-sample)
+                  (set (map :_source res))))
+           (doseq [elem res]
+             (base-check elem))))))))
+
+(deftest bulk-delete-update-test
+  (es-helpers/for-each-es-version
+      "bulk-delete and bulk-update shall properly handle authorization and not-found errors"
+      [5 7]
+    nil
+    (helpers/fixture-ctia-with-app
+     (fn [app]
+       (let [conn-state (get-conn-state app :sighting)
+             base-sighting (assoc sighting-minimal :tlp "amber")
+             total 10
+             sighting-ids (map #(str "sighting-" %) (range total))
+             sightings (map #(assoc base-sighting :id %) sighting-ids)
+             with-owner (fn [{:keys [login groups]} doc]
+                             (assoc doc
+                                    :owner login
+                                    :groups groups))
+             ident1 {:login "john-doe"
+                     :groups ["group1"]}
+             ident2 {:login "jane-doe"
+                     :groups ["group2"]}
+
+             [docs-1 docs-2 docs-3] (partition-all (/ total 3) sightings)
+             docs1-ident1 (map (partial with-owner ident1) docs-1)
+             docs2-ident2 (map (partial with-owner ident2) docs-2)
+             docs3-ident1 (map (partial with-owner ident1) docs-3)
+             bulk-update-handler (sut/bulk-update ss/Sighting)
+
+             _ (create-fn conn-state
+                       (concat docs1-ident1 docs2-ident2)
+                       ident
+                       {})
+             to-update-docs (map #(assoc % :title "updated title")
+                                 (concat docs1-ident1 docs2-ident2 docs3-ident1))
+             update-res (bulk-update-handler conn-state
+                                             to-update-docs
+                                             ident1
+                                             {})
+             delete-res (sut/bulk-delete conn-state
+                                         sighting-ids
+                                         ident1
+                                         {})
+             expected-not-found (set (concat (map :id docs-3)
+                                             (map :id docs-2)))
+             expected-delete-result {:deleted (set (map :id docs-1))
+                                     :errors {:not-found expected-not-found}}
+             expected-update-result {:updated (set (map :id docs-1))
+                                     :errors {:not-found expected-not-found}}]
+         (is (= expected-update-result
+                (-> (update-in update-res [:errors :not-found] set)
+                    (update :updated set))))
+         (is (= expected-delete-result
+                (-> (update-in delete-res [:errors :not-found] set)
+                    (update :deleted set)))))))))
```

Comment:
I think that if you want to test how `bulk-update` and `bulk-delete` behave together you probably want to test more scenarios.

1. (this one) generate n docs => check you can update then delete them all
2. generate n docs => update the 1->k and delete all even ones
3. create a doc, update it, delete it, try to re-update it, try to re-delete it
4. create a doc, concurrently update it and delete it
5. same as 2 but concurrently, would there always be half entities? Does some update could occurs concurrently and thus have any number of entites from n/2 to n? 

So you could expose some properties to provide the correct expectations (does delete enforce a 100% certainty of deletion or not).

Reply:
I added a comment in the bulk update task, I prefer to limit the inflation of this PR :-)
https://github.com/advthreat/iroh/issues/2249#issuecomment-895908314